src/utils/axiosSetup.ts

```diff
@@ -1,65 +1,91 @@
-        if (error.response.status === 401 && !untouchedRequest._retry) {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-          untouchedRequest._retry = true;
+        const toThrow: ErrorFormat = {
+          statusCode: errorAsAxiosError.code,
+          message: errorAsAxiosError.message,
+          raw: error,
+        };
 
-          // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-          if (!apiInstanceData.state.refreshingToken) {
-            apiInstanceData.state.refreshingToken = apiInstanceData.authentication.refreshToken();
-          }
+        throw toThrow;
+      }
 
-          const res = await apiInstanceData.state.refreshingToken;
-          apiInstanceData.state.refreshingToken = false;
-
-          if (res.status === 200 && res.data.access_token) {
-            const { access_token: accessToken, refresh_token: refreshToken } = res.data;
-            apiInstanceData.authentication.setTokens({ accessToken, refreshToken });
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            untouchedRequest.headers.Authorization = `Bearer ${accessToken}`;
-            if (apiInstanceData.config.settings.onTokenRefreshedCallback) {
-              apiInstanceData.config.settings.onTokenRefreshedCallback({ accessToken, refreshToken });
-            }
-            return axios(untouchedRequest);
-          } else if (apiInstanceData.config.settings.onTokenRefreshedFailed) {
-            apiInstanceData.config.settings.onTokenRefreshedFailed();
-          }
+      interceptorParams.state.refreshingToken = false;
+
+      if (res.status === 200 && res.data.access_token) {
+        const { access_token: accessToken, refresh_token: refreshToken } = res.data;
+        interceptorParams.authentication.setTokens({ accessToken, refreshToken });
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        untouchedRequest.headers.Authorization = `Bearer ${accessToken}`;
+        if (interceptorParams.config.settings.onTokenRefreshedCallback) {
+          interceptorParams.config.settings.onTokenRefreshedCallback({ accessToken, refreshToken });
         }
+        return axios(untouchedRequest);
       }
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/strict-boolean-expressions
-      throw error.response || error;
     }
-  );
+  }
+
+  const toThrow: ErrorFormat = {
+    statusCode: error.code != null && error.code.length > 0 ? parseInt(error.code) : undefined,
+    message: error.message,
+    raw: error,
+  };
+
+  throw toThrow;
+};
+
+export const interceptor = (interceptorParams: InterceptorParams): void => {
+  axios.interceptors.request.use(requestInterceptor);
+  axios.interceptors.response.use(responseFulfilledInterceptor, responseRejectedInterceptor(interceptorParams));
```

Comment:
One of those is not an interceptor. One of those takes a param and returns an interceptor.